core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -418,6 +437,36 @@ private[spark] class ExecutorMonitor(
     }
   }
 
+  override def onExecutorExcluded(executorExcluded: SparkListenerExecutorExcluded): Unit = {
+    if (!conf.get(EXCLUDE_ON_FAILURE_KILL_ENABLED)) {
+      val exec = executors.get(executorExcluded.executorId)
+      exec.isExcluded = true
```

Comment:
Here and elsewhere, defend against `exec` being `null` ?
Either with explicit `if (null != exec)` check or via `Option(executors.get(...)).foreach(exec => )`